sqlite3/lib/src/api/functions.dart

```diff
@@ -3,7 +3,7 @@ import 'package:meta/meta.dart';
 /// A scalar function exposed to sql.
 ///
 /// {@template sqlite3_function_behavior}
-/// The function must either return a `bool`, `num`, `String`, `List<int>` or
+/// The function must either return a `bool`, `num`, `String`, `Uint8List` or
```

Comment:
I don't think we should force users to return a `Uint8List` here. It's a breaking change which IMO is not necessary. Most users return a `Uint8List` already, I don't see much value in an enforcement in the type system.

We can check if we got an `Uint8List` and implicitly copy to an `Uint8List` if we didn't.